### PR TITLE
fix(ui): scale the doughnut chart with the font size

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Update the doughnut chart to fit within its container.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to a lxd pod details page that shows a doughnut chart.
- Change your browser's font size to something smaller than the default.
- The doughnut chart should resize and not overlap the content next to it.

## Fixes

Fixes: canonical-web-and-design/maas-ui#1588

### CODE REVIEW

File: ui/src/app/base/components/DoughnutChart/DoughnutChart.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const DoughnutChart = ({
     []
   );
 
-  const id = `doughnut-chart-${nanoid()}`;
+  const id = useRef(`doughnut-chart-${nanoid()}`);
```
Comment (huwshimi) [inline ui/src/app/base/components/DoughnutChart/DoughnutChart.tsx:70]: Drive-by to fix the id being updated on every render.

File: ui/src/app/base/components/DoughnutChart/DoughnutChart.tsx [hunk 2]
```diff
@@ -209,14 +200,7 @@ export const DoughnutChart = ({
             height={canvasSize}
             fill="transparent"
           />
-          {/* Move the chart on the canvas to leave enough space so that the hovered segments don't get cut off. */}
-          <g
-            transform={`xtranslate(${adjustedHoverWidth / 2}, ${
```
Comment (huwshimi) [inline ui/src/app/base/components/DoughnutChart/DoughnutChart.tsx:214]: Drive-by to remove a disabled transform.

**Review by Caleb-Ellis — DISMISSED** [review]
Nice! Looks good in FF and Chrome. :mag:

**Review by Caleb-Ellis — CHANGES_REQUESTED** [review]
Sorry, spoke too soon. The font-size bug is fixed, but the chart looks massive on smaller screens. Can we just set a max-width on the parent container?

**Comment by huwshimi** [comment]
> Sorry, spoke too soon. The font-size bug is fixed, but the chart looks massive on smaller screens. Can we just set a max-width on the parent container?
> 

Whoops, yep, fixed.

**Review by Caleb-Ellis — APPROVED** [review]
LGTM!

**Pull request merged**